### PR TITLE
Refactor main-map; fix routing bug with redirect

### PR DESCRIPTION
**Pull request by allthesignals**
Addresses #805

- This PR fixes a bug where direct, query-param'd links to main root of the app were redirecting to /about _without_ query params. Please review`services/layer-groups.js`
- Tests are added to demonstrate bug and increase coverage in main-map component
- As a rider, this PR splits out the measurement tools part of the map - specifically, that component is called map-measurement-tools. It shouldn't receive a thorough code review as it's largely moving code around. I would like input on the general direction (splitting it out into a component, the name of that component, using contextual params, etc).

### CODE REVIEW

**Comment by julialucyhogan** [comment]
Is this another case where you don't want to refactor the internals of the new component until there are tests, or should I be doing a real code review on the content of the component?

**Comment by allthesignals** [comment]
> Is this another case where you don't want to refactor the internals of the new component until there are tests, or should I be doing a real code review on the content of the component?

Thanks, I'm avoiding a refactor of the internals of the new component until I can be more confident following more test coverage...

File: app/services/layer-groups.js [hunk 1]
```diff
@@ -2,6 +2,16 @@ import Service from '@ember/service';
...
 import { copy } from 'ember-copy';
 
+const areIncomingLayerGroupsDefault = function(layerGroupsToShow, layerGroupsByDefault) {
+  const layersToShowAreDefault = layerGroupsToShow.every(
+    (layerGroupToShow, idx) => layerGroupToShow === layerGroupsByDefault[idx],
+  );
+
+  const currentDiffersFromDefault = layerGroupsToShow.length === layerGroupsByDefault.length;
+
+  return layersToShowAreDefault && currentDiffersFromDefault;
```
Comment (trbmcginnis) [inline app/services/layer-groups.js]: The naming here is great/intuitive, but I would love some comments explaining why this change was needed.

A little confused on why `currentDiffersFromDefault` would equal visible layer-groups having the same length as the default ones. Wouldn't that be the opposite?
